### PR TITLE
[#137] Remove `checkLocalhost` option

### LINKED ISSUE #137: Remove `checkLocalhost` option
# Clarification and motivation

In #85, we added the `checkLocalhost` option to decide whether to verify links to localhost.

However, upon further reflection, it seems like this could have been subsumed by the existing `ignoreRefs` option instead.

We should just add the regex `^https?://localhost` (and a similar one for `127.0.0.1`) to `ignoreRefs`'s default list.

# Acceptance criteria

* The `checkLocalhost` option has been removed
* We still ignore localhost by default, because it's in the `ignoreRefs` by default.

### PR DESCRIPTION
**Pull request by Sereja313**
## Description

Problem: In #85, we added the `checkLocalhost` option to decide whether to verify links to localhost. However, upon further reflection, it seems like this could have been subsumed by the existing `ignoreRefs` option instead.

Solution: Remove `check-localhost` CLI option and `checkLocalhost` config option. Add a regex matching localhost links to the `ignoreRefs` field of the default config.
## Related issue(s)

Fixes #137 

## :white_check_mark: Checklist for your Pull Request

Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](/README.md)
    - Haddock

- Public contracts
  - [ ] Any modifications of public contracts comply with the [Evolution
  of Public Contracts](https://www.notion.so/serokell/Evolution-of-Public-Contracts-2a3bf7971abe4806a24f63c84e7076c5) policy.
  - [x] I added an entry to the [changelog](/CHANGES.md) if my changes are visible to the users
        and
  - [ ] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [ ] My code complies with the [style guide](/docs/code-style.md).

### CODE REVIEW

File: tests/golden/check-cli/config-no-scan-ignored.yaml [hunk 1]
```diff
@@ -11,7 +11,6 @@ verification:
   notScanned: [ "to-ignore/broken-link.md" ]
   virtualFiles: []
   ignoreRefs: []
-  checkLocalhost: false
```
Comment (Sorokin-Anton) [inline tests/golden/check-cli/config-no-scan-ignored.yaml:14]: checkLocahost was false, so we should add anti-localhost regex?

**Review by Sorokin-Anton — COMMENTED** [review]
Looks good (except one yaml file with my comment)

File: tests/golden/check-cli/config-no-scan-ignored.yaml (hunk 1 above)
Comment (Sorokin-Anton) [reply tests/golden/check-cli/config-no-scan-ignored.yaml:14]: There is no localhost links in file, so this is not important, i think

**Review by Sorokin-Anton — APPROVED** [review]

File: tests/golden/check-localhost/check-localhost.bats [hunk 2]
```diff
@@ -8,30 +8,15 @@ load '../helpers/bats-support/load'
 load '../helpers/bats-assert/load'
 load '../helpers'
 
-@test "Config: check-localhost=false, CLA --check-localhost not provided" {
+@test "Ignore localhost" {
```
Comment (dcastro) [inline tests/golden/check-localhost/check-localhost.bats:11]: I guess `check-ignoreRefs` would now be a more appropriate name for this folder + bats file :thinking:

**Review by dcastro — APPROVED** [review]
LGTM, just a couple of small comments but pre-approving anyway :smile:

**Pull request merged**